src/network/networkd-route.c

```diff
@@ -516,6 +518,65 @@ int route_expire_handler(sd_event_source *s, uint64_t usec, void *userdata) {
         return 1;
 }
 
+static int append_nexthop_one(Route *route, MultipathRoute *m, struct rtattr *rta, size_t max_length, struct rtnexthop *rtnh) {
+        int r;
+
+        *rtnh = (struct rtnexthop) {
+                .rtnh_len = sizeof(*rtnh),
+                .rtnh_ifindex = m->ifindex,
+                .rtnh_hops = m->weight > 0 ? m->weight - 1 : 0,
+        };
```

Comment:
hmm, I am generally not a fan of functions that modify parameters on failure. Can't we find a nicer way for ths? i.e. isn't it enough to move these assignments to the end of this function when we know that everything worked correctly?